spec/lib/tasks/create_provider_partnerships_spec.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require 'rake'
+
+describe 'publish_give_course' do
+  subject do
+    Rake::Task['provider_partnerships:create_from_enrichments'].invoke
+  end
+
+  let!(:accredited_provider) { create(:accredited_provider) }
+  let!(:description) { 'The Description' }
+  let!(:training_provider) { create(:provider, accrediting_provider_enrichments: [AccreditingProviderEnrichment.new(UcasProviderCode: accredited_provider.provider_code, Description: description)]) }
+
+  Rails.application.load_tasks if Rake::Task.tasks.empty?
+
+  it 'calls Courses::PublishService service' do
+    expect { subject }.to change(training_provider.accredited_partnerships, :count).by(1)
+  end
```

Comment:
The description and the test don't match. It tests if partnerships of `training_provider` are increased by `1`.

I'd check if the accredited provider matches in the test too.